### PR TITLE
mgr: Define PY_SSIZE_T_CLEAN ahead of every Python.h

### PR DESCRIPTION
**Pull request by zaitcev**
Building on Fedora 35 with Python 3.10 makes vstart to loop
forever, throwing the following message:

 Error EINVAL: SystemError: PY_SSIZE_T_CLEAN macro must be
 defined for '#' formats

I followed the hint in the following document:
 https://docs.python.org/3/c-api/intro.html

It says "recommended" to always define PY_SSIZE_T_CLEAN,
but as you can see it is actually required in our case.

This version implements Kefu's suggestion to define the macro
on the command line, rather than insert it into every file.

Fixes: https://tracker.ceph.com/issues/53441
Signed-off-by: Pete Zaitcev <zaitcev@redhat.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
```diff
diff --git a/src/mgr/CMakeLists.txt b/src/mgr/CMakeLists.txt
index 55147af4fc6..b46e24b1847 100644
--- a/src/mgr/CMakeLists.txt
+++ b/src/mgr/CMakeLists.txt
@@ -33,6 +33,8 @@ if(WITH_MGR)
     mgr_commands.cc
     $<TARGET_OBJECTS:mgr_cap_obj>)
   add_executable(ceph-mgr ${mgr_srcs})
+  target_compile_definitions(ceph-mgr
+    PRIVATE PY_SSIZE_T_CLEAN)
   if(WITH_LIBCEPHSQLITE)
     target_link_libraries(ceph-mgr cephsqlite SQLite3::SQLite3)
   endif()
```
@zaitcev how about adding this definition using CMake? simpler and less error-prone this way.

**Comment by tchaikov** [comment]
also, might want to s/python/mgr/ in the title of the commit message.

File: src/ceph_mgr.cc
Comment (tchaikov) [inline src/ceph_mgr.cc:17]: we don't need this if this macro is already defined in the cmake script.

**Review by tchaikov — APPROVED** [review]
lgtm modulo a small nit.

File: src/ceph_mgr.cc
Comment (zaitcev) [reply src/ceph_mgr.cc:17]: Sorry, I see it now. In src/mgr/CMakeLists.txt, the src/ceph_mgr.cc is added to the list with the whole path ${CMAKE_SOURCE_DIR}/src/ceph_mgr.cc.

**Review by tchaikov — APPROVED** [review]

**Comment by eqrx** [comment]
Hey all, and thank you for the PR @zaitcev. I believe this also broke the current version of the ceph package in archlinux (see https://bugs.archlinux.org/task/73335?project=5&string=ceph).

I would assume no ... but is anyone aware of a workaround until this is merged and released that does not involve rebuilding ceph for myself? Thanks!

**Pull request merged**
